Book-A-Desk.Domain/Reservation.Validator.fs

```diff
@@ -2,19 +2,23 @@
 
 open System
 
+open System.ComponentModel.DataAnnotations
+open System.Text.RegularExpressions
 open Book_A_Desk.Core
-open Book_A_Desk.Domain
 open Book_A_Desk.Domain.Office.Domain
 open Book_A_Desk.Domain.Reservation.Commands
 
 module BookADeskReservationValidator =
     
-    let private validateEmailIsNotEmpty email =
-        match email with
-        | EmailAddress "" ->
-            Error "The e-mail address must not be empty."
-        | _ ->
-            Ok ()
+    let private validateCorporateEmail email =
+        let emailToValidate = email.ToString().ToLower()
+        let emailValidator = EmailAddressAttribute()
+        let isValidEmail = emailValidator.IsValid(emailToValidate)
+        let hasCorporateDomain = Regex.Match(emailToValidate, "@broadsign.com")
```

Comment:
this project is open source so we should probably inject the corporate domain

Reply:
+1

Reply:
Good point. Feel free to update the PR

Reply:
@jimmy-cyr I will take care, good point.

Reply:
I just pushed the commit to implement that.